src/algebra/module/basic.lean

```diff
@@ -413,7 +416,10 @@ namespace int
 variables [semiring R] [add_comm_group M] [semimodule R M]
 
 instance smul_comm_class : smul_comm_class ℤ R M :=
-{ smul_comm := λ z r l, by cases z; simp [←gsmul_eq_smul, ←nat.smul_def, smul_comm] }
+{ smul_comm := λ z r l, by cases z; simp [←gsmul_eq_smul, ←nat.smul_def, smul_comm (_ : ℕ)] }
+
+-- `smul_comm_class.symm` is not registered as an instance, as it would cause a loop
```

Comment:
This comment is extremely confusing.

Reply:
Would this be better as:
```suggestion
-- `_root_.smul_comm_class.symm` is not registered as an instance, as it would cause a loop
```

Reply:
Maybe
```suggestion
-- `smul_comm_class.symm` is not registered as an instance for arbitrary semirings, as it would cause a loop
```

Reply:
That's not accurate though @fpvandoorn, `smul_comm_class` doesn't require semirings.

Is this better?
```
-- `smul_comm_class.symm` is not registered as an instance automatically, as it
-- would cause a loop with itself; so we must register the specialization here.
```

Reply:
I could also just remove this comment, we don't have one at the other call site:

https://github.com/leanprover-community/mathlib/blob/6548be43c353eadb4df18d3f200034bb14374c07/src/algebra/algebra/basic.lean#L1201-L1203